pydoctor/templatewriter/pages/__init__.py

```diff
@@ -30,34 +31,125 @@ def signature(function: model.Function) -> str:
     return str(function.signature)
 
 class DocGetter:
-    def get(self, ob, summary=False):
+    """L{epydoc2stan} bridge."""
+    def get(self, ob:model.Documentable, summary:bool=False) -> Tag:
         if summary:
             return epydoc2stan.format_summary(ob)
         else:
-            doc = epydoc2stan.format_docstring(ob)
-            typ = epydoc2stan.type2stan(ob)
-            if typ is None:
-                return doc
+            return epydoc2stan.format_docstring(ob)
+    def get_type(self, ob: model.Documentable) -> Optional[Tag]:
+        return epydoc2stan.type2stan(ob)
+
+class BaseElement(Element, abc.ABC):
+    """
+    Common base HTML element. 
+    """
+    def __init__(self, 
+        system:Optional[model.System]=None, 
+        template_lookup:Optional[TemplateLookup]=None, 
+        loader:Optional[ITemplateLoader]=None, ) -> None:
+        """
+        Init a new pydoctor HTML element. 
+
+        The C{loader} property is usually got from the L{TemplateLookup} 
+        object but can also be set by argument.
+
+        @raises RuntimeError: If not enought information is provided 
+            to create the template loader. i.e. missing C{template_lookup} or C{loader} argument.
+            Could also be that the L{Template.renderable} property is not a L{ITemplateLoader} provider. 
+
+        @note: C{system} and C{template_lookup} can be none in special cases, like for L{LetterElement}. 
+        """
+        self.system = system
+        self.template_lookup = template_lookup
+        
+        if loader:
+            pass
+        # filename attribute should be set for most page classes
+        elif self.filename:
+            if self.template_lookup:
+                template = self.template_lookup.get_template(
+                    self.filename)
+                if not zope.interface.verify.verifyObject(ITemplateLoader, template.renderable):
+                    raise RuntimeError(f"Cannot create HTML element {self} because template renderable "
+                                       f"property is not a ITemplateLoader provider: {type(template.renderable)}")
+                else:
+                    loader = template.renderable
             else:
-                return [doc, ' (type: ', typ, ')']
+                raise RuntimeError(f"Cannot create HTML element {self} because no TemplateLookup "
+                                    "object is passed to BaseElement's 'template_lookup' init argument.")
+        else:
+            raise RuntimeError(f"Cannot create HTML element {self} because no ITemplateLoader "
+                                "object is passed to BaseElement's 'loader' init argument and 'filename' property is not set.")
+        super().__init__(loader)
+
+    @abc.abstractproperty
+    def filename(self) -> str:
+        """
+        Associated output filename. 
```

Comment:
Is "output" correct? I'd expect pages to have an output filename, but not elements. Also it seems this is called to get the template filename.